src/dual.jl

```diff
@@ -609,6 +609,14 @@ end
     return (Dual{T}(sd, cd * partials(d)), Dual{T}(cd, -sd * partials(d)))
 end
 
+# sincospi #
+#----------#
+
+@inline function Base.sincospi(d::Dual{T}) where T
+    sd, cd = sincospi(value(d))
+    return (Dual{T}(sd, cd * π * partials(d)), Dual{T}(cd, -sd * π * partials(d)))
```

Comment:
```suggestion
    return (Dual{T}(sd, cd * T(π) * partials(d)), Dual{T}(cd, -sd * T(π) * partials(d)))
```
Do we need to wrap in `T` to be type stable?

Reply:
The existing version should be fine since the type of `cd * π` will be decided by `cd`.

Reply:
Ah, that's awesome. So there are special promotion rules with literals?

Reply:
It's special `Irrational` magic.
```julia
julia> typeof(π)
Irrational{:π}
```
The precision of them in binary operations is determined by the other argument.